src/components/Vector3Input.js

```diff
@@ -7,11 +7,11 @@ export default function Vector3Input({ value, onChange, ...rest }) {
   return (
     <div className={styles.vector3Input}>
       <div className={styles.label}>X:</div>
-      <NumericInput value={value.x} onChange={x => onChange("x", x)} {...rest} />
+      <NumericInput value={value.x} onChange={x => onChange({ x, y: value.y, z: value.z })} {...rest} />
       <div className={styles.label}>Y:</div>
-      <NumericInput value={value.y} onChange={y => onChange("y", y)} {...rest} />
+      <NumericInput value={value.y} onChange={y => onChange({ x: value.y, y, z: value.z })} {...rest} />
```

Comment:
Check the key value pairs in your calls to onChange. x,y, and z don't all match up.

Reply:
Ah, good catch.